### DISCUSSION TITLE
When self-hosting, there are requirements for the database version

### DISCUSSION THREAD

**Discussion by s0214lx** [windmill-labs/windmill#4945] [Questions]
When self-hosting, set the local postgresql database. Is there any requirement for the database version? I use postgresql12 and get an error. ：
windmill_server-1 | Error: Migrating database: while executing migration 20231002123723: error returned from database: Syntax error in "TRIGGER" or nearby

**Comment by rubenfiszel** [comment]
you need pg14+
